### PR TITLE
chore: cleanup unused uploadMiddleware

### PR DESCRIPTION
**Pull request by nickskalkin**
This PR resolves [ONYX-504]

### Description

Onyx is cleaning up code related to the TinEye reverse image search project. Mostly the code was already removed by @gkartalis here https://github.com/artsy/eigen/pull/8898, but there were some leftovers introduces in these PRs:

- https://github.com/artsy/eigen/pull/6690 
- https://github.com/artsy/eigen/pull/6729

### PR Checklist

- [x] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

- remove unused uploadMiddleware - @nickskalkin 

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[ONYX-504]: https://artsyproduct.atlassian.net/browse/ONYX-504?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/app/system/relay/defaultEnvironment.ts [hunk 1]
```diff
@@ -17,7 +17,6 @@ import {
 import { rateLimitMiddleware } from "./middlewares/rateLimitMiddleware"
 import { simpleLoggerMiddleware } from "./middlewares/simpleLoggerMiddleware"
 import { timingMiddleware } from "./middlewares/timingMiddleware"
-import { uploadMiddleware } from "./middlewares/uploadMiddleware"
```
Comment (MounirDhahri) [inline src/app/system/relay/defaultEnvironment.ts:20]: **Q:** Should we revert to using the default `uploadMiddleware` from `react-relay-network-modern/node8` as before we rolled out this [here](https://github.com/artsy/eigen/pull/6729/files#diff-3269c086c6a1034aaf84e7624451de6c752ba88022a10be97f3e60ea9bb1572e)

Comment (nickskalkin) [reply src/app/system/relay/defaultEnvironment.ts:20]: Ooops, you are right, on it!

**Review by MounirDhahri — APPROVED** [review]
Nice - LGTM

**Review by brainbicycle — APPROVED** [review]
thanks for following up on this ! 🙌

**Pull request merged**